images/win/scripts/Tests/MSYS2.Tests.ps1

```diff
@@ -1,23 +1,64 @@
-BeforeAll {
-    $msys2BinDir = "C:\msys64\usr\bin"
-    $msys2mingwDir = "C:\msys64\mingw64\bin"
-}
+$toolsetContent = (Get-ToolsetContent).MsysPackages
+$archs = $toolsetContent.mingw.arch
+
+Describe "MSYS2 packages" {
+    BeforeAll {
+        $msys2Dir = "C:\msys64\usr\bin"
+    }
 
-Describe "MSYS2" {
-    It "<ToolName>" -TestCases @(
+    It "msys2Dir exists" {
+        $msys2Dir | Should -Exist
+    }
+    
+    $TestCases = @(
         @{ ToolName = "bash.exe" }
         @{ ToolName = "tar.exe" }
         @{ ToolName = "make.exe" }
-    ) {
-        Join-Path $msys2BinDir $ToolName | Should -Exist
+    )
+
+    It "<ToolName> is installed in <msys2Dir>" -TestCases $TestCases {
+        $env:PATH = "$msys2Dir;$env:PATH"
+        (Get-Command "$ToolName").Source | Should -BeLike "$msys2Dir*"
+    }
+
+    It "<ToolName> is avaialable" -TestCases $TestCases {
+        "$ToolName" | Should -ReturnZeroExitCodeWithParam
     }
+}
+
+foreach ($arch in $archs) {
+    Describe "$arch arch packages" {
+        $archPackages = $toolsetContent.mingw | Where-Object { $_.arch -eq $arch }
+        $tools = $archPackages.runtime_packages.name
+
+        foreach ($tool in $tools) {
+            Context "$tool package"{
+                if ($arch -eq "mingw-w64-i686")
+                {
+                    $ExecDir = "C:\msys64\mingw32\bin"
+                }
+                else
+                {
+                    $ExecDir = "C:\msys64\mingw64\bin"
+                }
+
+                $executables = ($archPackages.runtime_packages | Where-Object { $_.name -eq $tool }).executables | ForEach-Object {
```

Comment:
nitpick: remove () and add `ExecName = $_.executables`

Reply:
@al-cheb unfortunately, the result will be different and pester won't able to process it correctly
![image](https://user-images.githubusercontent.com/48208649/122439714-8e110b80-cfa4-11eb-8553-2cfcc1feaad0.png)
![image](https://user-images.githubusercontent.com/48208649/122439787-9d905480-cfa4-11eb-993e-f65a8ee25ddb.png)